### PR TITLE
feat(fxsql): Provided module

### PR DESCRIPTION
**Pull request by ekkinox**
FxSQL module :
- sql.DB built from config and Fx injection 
- hooks
- migrator
- seeder

### CODE REVIEW

File: fxsql/go.mod [hunk 1]
```diff
@@ -0,0 +1,69 @@
+module github.com/ankorstore/yokai/fxsql
```
Comment (ndk) [inline fxsql/go.mod:1]: Looks like there're no changes. EOL?

File: fxsql/.golangci.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+run:
```
Comment (ndk) [inline fxsql/.golangci.yml:1]: No changes as well

**Review by ndk — COMMENTED** [review]
LGTM

**Review by ndk — APPROVED** [review]

File: fxsql/README.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+```
+
+You can then create a migration file in this path.
+
+For example, `db/migrations/00001_create_foo_table.sql`:
+
+```sql
+-- +goose Up
+CREATE TABLE foo (
+    id   INTEGER NOT NULL PRIMARY KEY,
```
Comment (Skyld) [inline fxsql/README.md]: nitpic: Indentation

File: fxsql/README.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+}
+
+func (h *ExampleHook) Before(ctx context.Context, event *sql.HookEvent) context.Context {
+	// before SQL operation logic
+
+	return ctx
+}
+
+func (h *ExampleHook) After(ctx context.Context, event *sql.HookEvent) {
+	// after SQL operation logic
```
Comment (Skyld) [inline fxsql/README.md:240]: Why not returning the context here as well, as in `Before()`? (genuine question)

**Review by Skyld — APPROVED** [review]
Just a few very minor comments, excellent work 🙌

File: fxsql/README.md (hunk 4 above)
Comment (ndk) [reply fxsql/README.md:240]: I think it's the most simple example. It's supposed to return a new context if you wanna add something to the context. :)

Comment (ekkinox) [reply fxsql/README.md:240]: it works this way inside any operation:

- ctx = hook.Before(ctx) // may extend the ctx (tracing for ex)
- base_operation(ctx)
- hook.After(ctx)

Nothing is ever done after hook.After(), so there is no point to return the ctx from there, nothing would use it.

Resolving conversation, feel free to reopen if needed :+1:

**Review by ndk — APPROVED** [review]

**Review by ndk — APPROVED** [review]

**Pull request merged**